### PR TITLE
Closes #7156: Update app links feature usage to include app links interceptor.

### LINKED ISSUE #7156: Update app links feature usage to include app links interceptor
### Why/User Benefit/User Problem
Android component's AppLinksFeature now uses the AppLinksInterceptor to first determine if an intent should be opened by a third party application.

### What/Requirements
Add the interceptor to Fenix implementation.

### Acceptance Criteria (how do I know when I’m done?)
AppLinksFeature and AppLinksInterceptor working correctly in Fenix

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/Services.kt [hunk 1]
```diff
@@ -42,6 +47,17 @@ class Services(
...
+    val appLinksInterceptor by lazy {
+        AppLinksInterceptor(
+            context,
+            interceptLinkClicks = true,
+            launchInApp = {
+                context.settings().preferences.getBoolean(
+                    context.getPreferenceKey(R.string.pref_key_open_links_in_external_app), false)
+            }
+        )
+    }
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/fenix/components/Services.kt]: Is that a `service`? 🤔

Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/components/Services.kt]: Not really, but I don't know a better place to put it.  The other interceptor current in use is the `FirefoxAccountsAuthFeature` which is just above this.

**Comment by sblatz** [comment]
@rocketsroger 


One last error looks like :)

e: /builds/worker/checkouts/src/app/src/main/java/org/mozilla/fenix/customtabs/CustomTabsService.kt: (12, 1): Class 'CustomTabsService' is not abstract and does not implement abstract base class member @Nullable protected/*protected and package*/ abstract fun extraCommand(@NonNull p0: String, @Nullable p1: Bundle?): Bundle? defined in mozilla.components.feature.customtabs.AbstractCustomTabsService

**Comment by rocketsroger** [comment]
> > e: /builds/worker/checkouts/src/app/src/main/java/org/mozilla/fenix/customtabs/CustomTabsService.kt: (12, 1): Class 'CustomTabsService' is not abstract and does not implement abstract base class member @nullable protected/_protected and package_/ abstract fun extraCommand(@nonnull p0: String, @nullable p1: Bundle?): Bundle? defined in mozilla.components.feature.customtabs.AbstractCustomTabsService

Yes, but this is not something I can fix in my change.  I've been trying to make sense for this error.

**Pull request merged**